### PR TITLE
Change the way how metric type is specified in metadata.yaml

### PR DESCRIPTION
**Pull request by dmitryax**
To simplify the logic and make it simpler for users to read what properties are accepted for specific types.

### CODE REVIEW

**Review by djaglowski — COMMENTED** [review]
I like this improvement. The code looks good.

There is a minor lint issue and also a unit test failure that is likely fixed by #6461.

File: cmd/mdatagen/loader.go [hunk 1]
```diff
@@ -50,15 +50,30 @@ type metric struct {
...
-	// Raw data that is used to set Data interface below.
-	YmlData *ymlMetricData `yaml:"data" validate:"required"`
-	// Date is set to generic metric data interface after validating.
-	Data MetricData `yaml:"-"`
+	// Sum stores metadata for sum metric type
+	Sum *sum `yaml:"sum"`
+	// Gauge stores metadata for gauge metric type
+	Gauge *gauge `yaml:"gauge"`
+	// Histogram stores metadata for histogram metric type
+	Histogram *histogram `yaml:"histogram"`
```
Comment (bogdandrutu) [inline cmd/mdatagen/loader.go:58]: What do you think between these 2:

```yaml
metrics:
  system.cpu.time:
    description: Total CPU seconds broken down by different states.
    unit: s
    data:
      sum:
        monotonic: true
        aggregation: cumulative
    attributes:
```

VS 

```yaml
metrics:
  system.cpu.time:
    description: Total CPU seconds broken down by different states.
    unit: s
    sum:
      monotonic: true
      aggregation: cumulative
    attributes:
```

**Review by bogdandrutu — APPROVED** [review]
Everything looks good, not sure though between the two proposals.

File: cmd/mdatagen/loader.go (hunk 1 above)
Comment (dmitryax) [reply cmd/mdatagen/loader.go:58]: The first one maybe more semantically correct, but I would lean towards better user experience here: 2.

**Pull request merged**